R/plumber.R

```diff
@@ -875,6 +812,29 @@ plumber <- R6Class(
     setErrorHandler = function(fun){
       private$errorHandler <- fun
     },
+    #' @description Add a function to customize what is returned in `$apiSpec()`.
+    #' @param api_fun This function should accept an OpenAPI Specification autogenerated by `plumber`.  It should return a list object in valid OpenAPI Specification format. This will not be validated.
+    setApiHandler = function(api_fun = force) {
+      # allow null values
+      if (!is.null(api_fun)) {
+        stopifnot(is.function(api_fun))
+      }
+      private$apiHandler <- api_fun
+    },
+    #' @description Set UI to use for API
+    #' @param enabled a logical value.
+    #' @param ui a character value. Default to `plumber.ui` option value.
+    #' @param ... Other params to be passed to ui functions.
+    setUI = function(enabled = TRUE, ui = getOption("plumber.ui", "swagger"), ...) {
```

Comment:
The `enabled` argument seems redundant. 
* If `enabled=FALSE`, `ui` is not used.
* If `enabled=TRUE`, `ui` is required.

I don't mind having the `enabled` value in the `private$ui_info`.  

```r
if (length(ui) == 1 && is.character(ui)) {
  enabled <- TRUE
} else {
  enabled <- FALSE
  ui <- "__not_enabled__"
}
```